internal/listener/listener.go

```diff
@@ -34,13 +34,26 @@ func NewListener(db *database.DB, runtimeConfig *config.RuntimeConfig, logs *log
 		logs:          logs,
 		runtimeConfig: runtimeConfig,
 	}
+
+	debugMux := http.NewServeMux()
+	debugMux.HandleFunc("/dump-config", l.DumpConfig)
+	debugMux.HandleFunc("/dump-incidents", l.DumpIncidents)
+	debugMux.HandleFunc("/dump-schedules", l.DumpSchedules)
+
+	l.mux.Handle("/debug/", http.StripPrefix("/debug", l.debugAuth(debugMux)))
 	l.mux.HandleFunc("/process-event", l.ProcessEvent)
-	l.mux.HandleFunc("/dump-config", l.DumpConfig)
-	l.mux.HandleFunc("/dump-incidents", l.DumpIncidents)
-	l.mux.HandleFunc("/dump-schedules", l.DumpSchedules)
 	return l
 }
 
+func (l *Listener) debugAuth(next http.Handler) http.Handler {
```

Comment:
Please add a short API doc header when introducing new methods (even if current methods are lacking one). In addition, the method name does not really represent what the method does. How about something like `requireDebugAuth`? It may be longer, but the name actually says what it does.

This may be nitpicky, but the position of the method in the file seems a bit off. In the current flow, I would expect it between `ProcessEvent` and `checkDebugPassword`.